packages/backend/src/ee/services/ai/agents/tests/agent.test.ts

```diff
@@ -0,0 +1,197 @@
+
+                    promptTestUtils.expectCorrectToolUsage(deps, [
+                        'getExplores',
+                        'getExplore',
+                        'searchFields',
+                    ]);
+                    promptTestUtils.expectResponseWithContent(firstResponse, [
+                        'customer id',
+                        'customer name',
+                        'customer email',
+                        'total revenue',
+                        'order count',
+                    ]);
+
+                    // 2. what are the top customers by revenue?
+                    const secondMessage = createMessage(
+                        'Can you show me the top customers by revenue?',
+                    );
+                    const conversationHistory = [
+                        firstMessage,
+                        createMessage(firstResponse, 'assistant'),
+                        secondMessage,
+                    ] satisfies ConversationHistory;
+                    const secondArgs = createArgs({
+                        messageHistory: conversationHistory,
+                    });
+
+                    const secondResponse = await generateAgentResponse({
+                        args: secondArgs,
+                        dependencies: deps,
+                    });
+
+                    promptTestUtils.expectResponseWithContent(secondResponse, [
+                        'chart',
+                        'top customers',
+                        'revenue',
+                    ]);
+                    promptTestUtils.expectCorrectToolUsage(deps, [
+                        'getExplores',
+                        'getExplore',
+                        'searchFields',
+                        'getPrompt',
+                    ]);
+
+                    // 3. give me that viz!
+                    const thirdMessage = createMessage(
+                        'Can you create a bar chart of this data?',
+                    );
+                    const fullConversation = [
+                        ...conversationHistory,
+                        createMessage(secondResponse, 'assistant'),
+                        thirdMessage,
+                    ] satisfies ConversationHistory;
+                    const thirdArgs = createArgs({
+                        messageHistory: fullConversation,
+                    });
+
+                    const thirdResponse = await generateAgentResponse({
+                        args: thirdArgs,
+                        dependencies: deps,
+                    });
+
+                    promptTestUtils.expectCorrectToolUsage(deps, [
+                        'getExplores',
+                        'getExplore',
+                        'searchFields',
+                        'getPrompt',
+                    ]);
+                    promptTestUtils.expectResponseWithContent(
+                        thirdResponse,
+                        'bar chart',
+                    );
```

Comment:
is this checking if text contains bar chart or not?

Reply:
Ideally, this should be a test case where you check that bar chart tool has been called

Reply:
updated!